docs/user_config.md

```diff
@@ -0,0 +1,150 @@
+# omitting the code from above Tiltfile that includes correct resources
+# Only configure the build for services we want to edit
+# Thanks to Tilt's existing behavior, an image that is mentioned in YAML but has no build set
+# will use the existing tag.
+to_edit = cfg.get('to-edit', [])
+if 'a' in to_edit:
+  docker_build('a', './a')
+if 'b' in to_edit:
+  docker_build('b', './b')
+if 'c' in to_edit:
+  docker_build('c', './c')
+if 'd' in to_edit:
+  docker_build('d', './d')
+```
+
+## The Config File
+In addition to config settings coming from command-line args, Tilt will read
+them from a `tilt_config.json` in the same directory as the `Tiltfile`, if one
+exists. It should contain a single JSON dict, where the keys are config setting
+names and the values are their values (e.g., `{"to-edit": ["b", "c"]}`).
+As with any other file read by the Tiltfile, if this config file is changed
+while Tilt is running, Tilt will pick up that change and reexecute the Tiltfile.
+If the same setting is specified in both the config file and the command-line
+args, the value from the args takes precedent.
+
+## Changing args at runtime
+While Tilt is running, if the user decides they need a different set of
+user args (i.e., args handled by config.parse, not args to tilt itself like
+`--hud`) than what they passed to `tilt up`, they can use `tilt flags` to
+replace them. `tilt flags --clear` tells Tilt to clear the set of user args in
+use.
+Note that settings from user args always take precedent over settings from the
+config file, even if the config file is changed after the user args were set.
+
+### Comparison to Default Behavior
+If you don't call `config.parse`, Tilt's default behavior is to set resources to
+any (non-empty) passed args. As a starting point, this is an equivalent Tiltfile
+implementation:
+```python
+config.define_string_list("args", args=True)
+cfg = config.parse()
+config.set_enabled_resources(cfg.get('args', []))
+```
+
+## Future Directions
+This section describes places we expect the config to go. Let us know if any of
+these would be particularly helpful to you and your team.
+### More Kinds of Settings
+You should be able to define more kinds of settings. For example, a bool, an
+an enum, or a singly-valued string. By moving error-checking to the built-in
+library, you can make your Tiltfile shorter and more correct.
+### Better Config Manipulation/Discovery
+Your users should be able to understand what reasonable values are. This could
+be from help output, tab-completion, or in the Web UI.
+Tilt should provide a way to manipulate tilt_config.json.
+Your users should be able to modify the config in the Web UI.
+### Escape Hatch for Argument Parsing
+You should be able to use a custom syntax for command-line arguments. The config
+module could take a function that would parse a command-line into JSON.
+### Interaction with load/include
+Tiltfiles that are used via `load`/`include` should be able to define and use
+config settigs.
```

Comment:
(mention the current behavior)

Reply:
you missed the typo!!!!

Reply:
Hmm, do you think that's important? I'd figured the obvious inference was "currently they cannot define and use config settings", but maybe that leaves open "well can they define config settings but not use them? use config settings but not define them?"

Reply:
(I ask because I think describing the current behavior would really overwhelm this bullet with cognitive load - you're not allowed to define new settings in any file after any file has called config.parse, and you're not allowed to call config.parse from multiple Tiltfiles in different directories. It seems more to the point just to give people the sense that they shouldn't go there)

Reply:
eh, fair point -- that's fine then, leave as is